packaging/resource_suite_s3_nocache.py

```diff
@@ -2198,6 +2198,98 @@ def test_iput_and_icp_with_invalid_secret_key__issue_6154(self):
+            self.admin.assert_icommand("iadmin addchildtoresc %s %s" % (repl_resource_name, s3_resc2_resource_name))
+            self.admin.assert_icommand("iadmin addchildtoresc %s %s" % (repl_resource_name, s3_resc3_resource_name))
+
+            file_sizes = [
+                    100,
+                    4 * 1024 * 1024 - 1,
+                    4 * 1024 * 1024,
+                    8 * 1024 * 1024,
+                    16 * 1024 * 1024,
+                    32 * 1024 * 1024 - 1,
+                    32 * 1024 * 1024,
+                    32 * 1024 * 1024 + 1,
+                    ];
+
+            for file_size in file_sizes:
+                for write_type in ['iput', 'istream']:
+                    # create the file
+                    s3plugin_lib.make_arbitrary_file(filename, file_size)
+
+                    # put the file to repl_resource
+                    if write_type == 'iput':
+                        self.user1.assert_icommand("iput -R %s %s" % (repl_resource_name, filename))
+                    else:  # istream
+                        buf = bytearray(file_size)
+                        with io.open(filename, 'rb') as fp:
+                            size = fp.readinto(buf)
+                            if not size:
+                                break
+                            buf_str = buf[:size].decode()
+                            logical_path = os.path.join(self.user1.session_collection, filename)
+                            self.user1.assert_icommand(['istream', 'write', '-R', repl_resource_name, logical_path], input=buf_str)
+
+                    # debug
+                    self.user1.assert_icommand("ils -l %s" % filename, 'STDOUT_SINGLELINE', filename)
+
+                    # get each replica
+                    self.user1.assert_icommand("iget -f -n 0 %s %s_get0" % (filename, filename))
+                    self.user1.assert_icommand("diff %s %s_get0" % (filename, filename), 'EMPTY')
+                    self.user1.assert_icommand("iget -f -n 1 %s %s_get1" % (filename, filename))
+                    self.user1.assert_icommand("diff %s %s_get1" % (filename, filename), 'EMPTY')
+                    self.user1.assert_icommand("iget -f -n 2 %s %s_get2" % (filename, filename))
+                    self.user1.assert_icommand("diff %s %s_get2" % (filename, filename), 'EMPTY')
+
+                    # cleanup before next loop
+                    s3plugin_lib.remove_if_exists(filename)
+                    s3plugin_lib.remove_if_exists("%s_get0" % filename)
+                    s3plugin_lib.remove_if_exists("%s_get1" % filename)
+                    s3plugin_lib.remove_if_exists("%s_get2" % filename)
+                    self.user1.assert_icommand("irm -f %s" % (filename))
+
+        finally:
+            # clean up the file that was put in case it wasn't cleaned up
+            try:
+                self.user1.assert_icommand("irm -f %s" % (filename))
```

Comment:
Is there a reason this is wrapped with the try-except? `irm -f` will not fail if the object does not exist, and if it fails for some other reason, then removing the resources below will fail as well.

Reply:
I didn't realize that it would not fail if the object does not exist.  I'll remove the try/except.